pkg/plugins/identity/oidc/provider.go

```diff
@@ -0,0 +1,174 @@
+func New(input *provider.PluginCreationInput) (identity.Provider, error) {
+	return &oidcIdentityProvider{
+		logger:      input.Logger,
+		interactive: input.IsInteractice,
+	}, nil
+}
+
+type oidcIdentityProvider struct {
+	logger      *zap.SugaredLogger
+	interactive bool
+}
+
+type providerConfig struct {
+	OidcServer string `json:"oidc-server"`
+	OidcId     string `json:"oidc-client-id"`
+	OidcSecret string `json:"oidc-client-secret"`
+	UsePkce    string `json:"oidc-use-pkce"`
+}
+
+func (p *oidcIdentityProvider) Name() string {
+	return ProviderName
+}
+
+// Authenticate will generate authentication config.
+func (p *oidcIdentityProvider) Authenticate(ctx context.Context, input *identity.AuthenticateInput) (*identity.AuthenticateOutput, error) {
+	p.logger.Info("using oidc for authentication")
+
+	p.getConfigFromUrl(input.ConfigSet)
+
+	cfg := &providerConfig{}
+	if err := config.Unmarshall(input.ConfigSet, cfg); err != nil {
+		return nil, fmt.Errorf("unmarshalling config into providerConfig: %w", err)
+	}
+
+	id := &oidc.Identity{
+		OidcServer: cfg.OidcServer,
+		OidcId:     cfg.OidcId,
+		OidcSecret: cfg.OidcSecret,
+		UsePkce:    cfg.UsePkce,
+	}
+
+	return &identity.AuthenticateOutput{
+		Identity: id,
+	}, nil
+}
+
+func (p *oidcIdentityProvider) getConfigFromUrl(configSet config.ConfigurationSet) {
+	if configSet.Get("config-url") != nil {
+		config := configSet.Get("config-url").Value
+		if config != nil {
+			configValue := config.(string)
+			if strings.HasPrefix(configValue, "http://") {
+				if configSet.Get("ca-cert") != nil {
+					caCert := configSet.Get("ca-cert").Value
+					if caCert != nil {
+						SetTransport(caCert.(string))
+					}
+				} else {
+					SetTransport("")
+				}
+
+				kclient := khttp.NewHTTPClient()
+				res, err := kclient.Get(configValue, nil)
+				if err == nil {
+					appConfiguration := &kconnectv1alpha.Configuration{}
+					if err := json.Unmarshal([]byte(res.Body()), appConfiguration); err == nil {
+						oidc := appConfiguration.Spec.Providers["oidc"]
+						for k, v := range oidc {
+							if k != "" && v != "" {
+								addItem(configSet, k, v)
+							}
+						}
+					} else {
+						p.logger.Errorf("Error loading payload from config URL, error is: %w", err)
+					}
+				} else {
+					p.logger.Errorf("Error calling config URL, error is: %w", err)
+				}
+			}
+		}
+	}
+
+}
+
+func addItem(configSet config.ConfigurationSet, key string, value string) {
+	if configSet.Exists(key) {
+		configSet.SetValue(key, value)
+	} else {
+		configSet.Add(
+			&config.Item{Name: key, Type: config.ItemType("string"), Value: value, DefaultValue: ""})
+	}
+}
+
+func SetTransport(file string) {
+
+	var config *tls.Config
+	if file != "" {
+		caCert, err := os.ReadFile(file)
+		if err != nil {
+			log.Fatal(err)
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+		config = &tls.Config{
+			RootCAs: caCertPool,
+		}
+	} else {
+		config = &tls.Config{InsecureSkipVerify: true}
```

Comment:
Hi @michael12312, what is the reason for disabling TLS certificate check?

Looks like GitHub is blocking the PR because of this issue, so might need to think of an alternative, or might need to investigate further if this is truly required.

Reply:
hi @Gearheads 
I think we need this part. Sometimes user would like to pass a cert file for SSL verification, but sometime they just want to skip this step, so add this option for them. 
Thanks!

Reply:
Ok sounds good. Thank you for the information.

Reply:
Maybe add flag to skip TLS check rather than making it as default behaviour.

Reply:
> Maybe add flag to skip TLS check rather than making it as default behaviour.

Yeah this would be great!

Reply:
hi @kishoregv 
I completely agree your comments.
Just got feedback from user that: most of the time they won't pass cert file, also want the less parameters the better, convenient for their use.
Do you think we can keep this? Avoid user passing 1 more parameter?

Reply:
I agree it is annoying to pass insecure true flag every time, reading insecure flag from config or environment variable makes it easier so user do not have to pass it every time.

I think making insecure default true is not a good idea.

Reply:
Hi @kishoregv 
Already made changes, now it’s required to pass skip-ssl flag, also can add this flag in config file, so no need to pass every time.
Also authentication and authorization part added, now user can got authenticated and authorized using Kconnect.
Please continue help to review, thanks very much!!

Reply:
Hi @michael12312, can we please remove this line?
```
config = &tls.Config{InsecureSkipVerify: true}
```
As mentioned in the GitHub error, we should not be setting this to true.

Reply:
Hi @Gearheads 

Considering sometime user may doesn't like to pass the cert file to kconnect everytime, so added this option for them.
Also this is not the default behavior, user needs to explicitly specify the flag, then will skip tls verify, for calling the config url.
Do you think we can keep this part?